testutils/src/main/scala/za/co/absa/enceladus/testutils/ComparisonJob.scala

```diff
@@ -19,47 +19,51 @@ import org.apache.spark.SparkContext
 import org.apache.spark.sql.types.{StructField, StructType}
 import org.apache.spark.sql.{DataFrameReader, Dataset, Row, SparkSession}
 
-object ComparisonJob {
-  def getReader(cmd: CmdConfig)(implicit sparkSession: SparkSession): DataFrameReader = {
-    val dfReader = sparkSession.read.format(cmd.rawFormat)
-    // applying format specific options
-    val  dfReader4 = {
-      val dfReader1 = if (cmd.rowTag.isDefined) dfReader.option("rowTag", cmd.rowTag.get) else dfReader
-      val dfReader2 = if (cmd.csvDelimiter.isDefined) dfReader1.option("delimiter", cmd.csvDelimiter.get) else dfReader1
-      val dfReader3 = if (cmd.csvHeader.isDefined) dfReader2.option("header", cmd.csvHeader.get) else dfReader2
-      dfReader3
-    }
-    if (cmd.rawFormat.equalsIgnoreCase("fixed-width")) {
-      val dfReader5 = if (cmd.fixedWidthTrimValues.get) dfReader4.option("trimValues", "true") else dfReader4
-      dfReader5
-    } else dfReader4
-  }
+import scala.annotation.switch
 
-  def removeMetadataFromSchema(schema: StructType): StructType = {
-    StructType(schema.map{ f =>
-      StructField(f.name, f.dataType, f.nullable)
-    })
-  }
+final case class CmpJobDatasetsDiffer(private val refPath: String,
```

Comment:
It's a best practice to specify suffix `Exception`. I would move both of these to their own package `exceptions`.

Reply:
TBH I don't share your point of view here. I am aware that it is something that persists in today's world, but it is in my opinion unnecessary to have. It should be readable from the name that it is an exception/error and when you get it in the stack trace you would expect it to be an exception, when you see the throw XY, you know it's an exception and lastly, when you see the declaration that extends Exception, you know what it is. These are 3 places you can find/work with exceptions and in no way, can it be mistaken for anything else.